designer/client/inline-conditions.js

```diff
@@ -0,0 +1,200 @@
+    return Object.assign(conditionalOperators[fieldType] || {}, conditionalOperators['default'])
+  }
+}
+
+class InlineConditions extends React.Component {
+  constructor (props) {
+    super(props)
+
+    const { data } = this.props
+
+    this.fields = [].concat.apply([],
+      data.pages.map(page => page.components.filter(component => component.name)
+        .map(component => ({ label: component.title, name: component.name, type: component.type, values: component.options.list ? data.lists.find(it => it.name === component.options.list).items : [] })))
+    ).reduce((obj, item) => {
+      obj[item.name] = item
+      return obj
+    }, {})
+
+    this.state = {
+      conditions: new ConditionsModel()
+    }
+  }
+
+  onClickAddItem = e => {
+    this.setState({
+      condition: {}
+    })
+  }
+
+  onClickAnd = e => {
+    this.setState({
+      condition: {
+        coordinator: 'and'
+      }
+    })
+  }
+
+  onClickOr = e => {
+    this.setState({
+      condition: {
+        coordinator: 'or'
+      }
+    })
+  }
+
+  onClickSave = e => {
+    const { conditions, condition } = this.state
+
+    this.setState({
+      conditions: conditions.add(new Condition(Field.from(condition.field), condition.operator, Value.from(condition.value), condition.coordinator)),
+      condition: undefined
+    })
+  }
+
+  conditionFieldId (id, fieldName) {
+    return `conditions.${id}.${fieldName}`
+  }
+
+  parseConditionId (fieldId) {
+    return fieldId.split('.')[1]
+  }
+
+  onChangeField = e => {
+    const input = e.target
+    const fieldName = input.value
+
+    const { condition } = this.state
+
+    const currentField = condition.field ? condition.field.name : undefined
+    const currentOperator = condition.operator
+
+    this._updateCondition(condition, c => {
+      if (fieldName) {
+        if (currentField && this.fields[currentField].values !== this.fields[fieldName].values) {
+          delete c.value
+        }
+        if (currentOperator && !conditionalOperators.getConditionals(fieldName)[currentOperator]) {
+          delete c.operator
+        }
+        c.field = new Field(fieldName, this.fields[fieldName].label)
+      } else {
+        delete c.field
+        delete c.operator
+        delete c.value
+      }
+    })
+  }
+
+  _updateCondition (condition, updates) {
+    const copy = clone(condition)
+    updates(copy)
+    this.setState({
+      condition: copy
+    })
+  }
+
+  onChangeOperator = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.operator = input.value })
+  }
+
+  onChangeValue = e => {
+    const input = e.target
+    const { condition } = this.state
+
+    this._updateCondition(condition, c => { c.value = new Value(input.value) })
+  }
+
+  render () {
+    const { conditions, condition } = this.state
+
+    const fieldDef = condition && condition.field ? this.fields[condition.field.name] : undefined
```

Comment:
are you relying on `undefined` value? 
`condition && condition.field && this.fields[condition.field.name]` will return the value of the last operand, otherwise will just return false. is falsey value ok for your code?

Reply:
good shout, done